civitai-downloader-v2/src/api/client.py

```
@@ -1,5 +1,167 @@
+                raise Exception(f"API error {response.status_code}: {response.text}")
+            
+            # Parse response
+            result = response.json()
+            
+            # Cache successful response
+            self.cache.store(cache_key, result)
+            
+            return result
+            
+        except httpx.TimeoutException:
+            raise Exception("Request timeout")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)